testplan/testing/multitest/driver/connection/base.py

```diff
@@ -0,0 +1,69 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Union
+from collections import defaultdict
+
+from testplan.common.utils.context import ContextValue
+
+
+class Direction(Enum):
+    CONNECTING = "connecting"
+    LISTENING = "listening"
+
+
+@dataclass
+class BaseConnectionInfo:
+    name: str  # name of the connection
+    service: str  # e.g. HTTP, TCP, FIX
+    protocol: str  # tcp, udp, file
+    identifier: Union[int, str, ContextValue]
+    direction: Direction
+
+    def to_dict(self):
+        return {
+            "service": self.service,
+            "protocol": self.protocol,
+            "identifier": self.identifier,
+            "direction": self.direction,
+        }
+
+    @property
+    def connection_rep(self):
+        raise NotImplementedError
+
+    def promote_to_connection(self):
+        raise NotImplementedError
+
+
+@dataclass
+class BaseDriverConnection:
+    """
+    Base class to show connection between drivers.
+    Each specific type (protocol) of connection should have its own subclass.
+    """
+
+    service: str
+    connection_rep: str
+    drivers_listening: "defaultdict[List]"
```

Comment:
`listening` tends to be meaningless for file-based conn, may have different attr for network & file instead of sharing the same in the base class